lib/pages/chat/events/download_video_widget.dart

```diff
@@ -0,0 +1,139 @@
+import 'package:fluffychat/pages/chat/events/download_video_state.dart';
+import 'package:fluffychat/pages/chat/events/event_video_player.dart';
+import 'package:fluffychat/pages/chat/events/message_content_style.dart';
+import 'package:fluffychat/presentation/mixins/handle_video_download_mixin.dart';
+import 'package:fluffychat/presentation/mixins/play_video_action_mixin.dart';
+import 'package:fluffychat/utils/localized_exception_extension.dart';
+import 'package:fluffychat/utils/platform_infos.dart';
+import 'package:fluffychat/utils/twake_snackbar.dart';
+import 'package:fluffychat/widgets/mxc_image.dart';
+import 'package:flutter/material.dart';
+import 'package:linagora_design_flutter/colors/linagora_ref_colors.dart';
+import 'package:matrix/matrix.dart';
+
+class DownloadVideoWidget extends StatefulWidget {
+  final Event event;
+
+  const DownloadVideoWidget({super.key, required this.event});
+
+  @override
+  State<StatefulWidget> createState() => _DownloadVideoWidgetState();
+}
+
+class _DownloadVideoWidgetState extends State<DownloadVideoWidget>
+    with HandleVideoDownloadMixin, PlayVideoActionMixin {
+  final _downloadStateNotifier = ValueNotifier(DownloadVideoState.initial);
+  String? path;
+  final downloadProgressNotifier = ValueNotifier(0.0);
+
+  @override
+  void initState() {
+    WidgetsBinding.instance.addPostFrameCallback((timeStamp) {
+      _downloadAction();
+    });
+    super.initState();
+  }
+
+  void _downloadAction() async {
+    _downloadStateNotifier.value = DownloadVideoState.loading;
+    try {
+      path = await handleDownloadVideoEvent(
```

Comment:
what happen user close the dialog during downloading time?

Reply:
it uses dio to download video, and dio is singleton, today I will find a way to cancel it when user exit the video

Reply:
https://github.com/linagora/twake-on-matrix/assets/43041967/e1ebb3ab-6d68-4009-a45e-b6790aa43781

Now user can cancel the download video